server/chat-plugins/wifi.ts

```diff
@@ -844,32 +844,30 @@ const cmds: Chat.ChatCommands = {
 		room = this.requireRoom('wifi' as RoomID);
 		this.checkCan('warn', null, room);
 
-		target = this.splitTarget(target, false);
-		const targetUser = this.targetUser;
-		if (!targetUser) return this.errorReply(`User '${this.targetUsername}' not found.`);
-		if (target.length > 300) {
+		const {targetUser, targetUsername, rest} = this.splitUser(target);
+		if (!targetUser) return this.errorReply(`User '${targetUsername}' not found.`);
+		if (rest.length > 300) {
 			return this.errorReply("The reason is too long. It cannot exceed 300 characters.");
 		}
-		if (Punishments.getRoomPunishType(room, this.targetUsername)) {
-			return this.errorReply(`User '${this.targetUsername}' is already punished in this room.`);
+		if (Punishments.getRoomPunishType(room, targetUsername)) {
+			return this.errorReply(`User '${targetUsername}' is already punished in this room.`);
 		}
 
-		Giveaway.ban(room, targetUser, target);
+		Giveaway.ban(room, targetUser, rest);
 		if (room.giveaway) room.giveaway.kickUser(targetUser);
-		this.modlog('GIVEAWAYBAN', targetUser, target);
-		if (target) target = ` (${target})`;
+		this.modlog('GIVEAWAYBAN', targetUser, rest);
+		if (rest) target = ` (${rest})`;
 		this.privateModAction(`${targetUser.name} was banned from entering giveaways by ${user.name}.${target}`);
```

Comment:
This is likely a bug. I'm guessing you used `target` instead of `rest` because `rest` is a const. But that introduces a bug when `rest` is empty.

(The correct fix would be to use `let` instead of `const`. It's fine if you don't change the other values!)

Reply:
idk what i did here, i definitely didnt intend to approach it like that